File: SNS.Web/src/app/jobs/jobs/contracts/job-details.dto.ts
import { CompanySnapshotDto } from '../../companies/contracts/company-snapshot.dto';
import { JobType } from '../../enums/job-type.enum';
import { SalaryType } from '../../enums/salary-type.enum';
import { JobSkillDto } from '../../job-skills/contracts/job-skill.dto';

export interface JobDetailsDto {
    id: string;
    title: string;
    description: string;
    companyId: string;
    company: CompanySnapshotDto;
    location: string;
    type: JobType;
    minSalary: number;
    maxSalary: number;
    currencyCode: string;
    salaryType: SalaryType;
    keyResponsibilitiesText: string;
    createdAt: string;
    updatedAt: string;
    closedAt: string | null;
    isActive: boolean;
    skills: JobSkillDto[];
}

File: backend/SNS.Application/Jobs/Jobs/Contracts/JobDetailsDto.cs
using SNS.Application.Jobs.Companies.Contracts;
using SNS.Domain.Jobs.Enums;

namespace SNS.Application.Jobs.Jobs.Contracts;

public sealed record JobDetailsDto(
    Guid Id,
    Guid CompanyId,
    CompanySnapshotDto Company,
    string Title,
    string Description,
    string Location,
    JobType Type,
    decimal MinSalary,
    decimal MaxSalary,
    string CurrencyCode,
    SalaryType SalaryType,
    string KeyResponsibilitiesText,
    List<string> Skills,
    int ApplicationsCount,
    bool HasApplied,
    DateTime CreatedAt,
    DateTime? ClosedAt,
    bool IsActive
);
